### PR TITLE
Pre calculated Machine energy profiles

### PR DESCRIPTION
**Pull request by ArneTR**

### CODE REVIEW

**Comment by ArneTR** [comment]
The overhead is now down from 23s to 5s

- Before: https://github.com/green-coding-solutions/eco-ci-energy-estimation/actions/runs/9511551439/job/26217894696
- After: https://github.com/green-coding-solutions/eco-ci-energy-estimation/actions/runs/9511695986/job/26218285479

It is unclear to me why the step for a measurement takes 5s. In local testing the time is less than 200ms

Even when I look at the detail logs the time seems to be less than 1s.

Screenshot 2024-06-14 at 8 30 26 AM

**Comment by ArneTR** [comment]
The overhead is now down from 23s to 5s

- Before: https://github.com/green-coding-solutions/eco-ci-energy-estimation/actions/runs/9511551439/job/26217894696
- After: https://github.com/green-coding-solutions/eco-ci-energy-estimation/actions/runs/9511695986/job/26218285479

It is unclear to me why the step for a measurement takes 5s. In local testing the time is less than 200ms

Even when I look at the detail logs the time seems to be less than 1s.

Screenshot 2024-06-14 at 8 30 26 AM

**Comment by ArneTR** [comment]
The overhead is now down from 23s to 5s

- Before: https://github.com/green-coding-solutions/eco-ci-energy-estimation/actions/runs/9511551439/job/26217894696
- After: https://github.com/green-coding-solutions/eco-ci-energy-estimation/actions/runs/9511695986/job/26218285479

It is unclear to me why the step for a measurement takes 5s. In local testing the time is less than 200ms

Even when I look at the detail logs the time seems to be less than 1s.

Screenshot 2024-06-14 at 8 30 26 AM

**Comment by ArneTR** [comment]
I found the culprit. Aparently sending a process to the background makes GitHub steps hang for a while of the input and output streams are not handled.

This code part fixes it: https://github.com/green-coding-solutions/eco-ci-energy-estimation/commit/4425e1bd2371ac042c38f61f2b784719919d4a15

Brought it now also to this branch

**Review by ribalba — COMMENTED** [review]
I really like the new solution. Makes the code a lot clearer and easier to maintain next to all the other advantages.

File: machine-power-data/default.sh [hunk 1]
```diff
@@ -0,0 +1,10005 @@
+#!/usr/bin/env bash
+set -eu
+declare -A cloud_energy_hashmap
+cloud_energy_hashmap[0.00]=93.3974838256836
```
[the remark this answers is not in the document]
Comment (ArneTR) [reply machine-power-data/default.sh:4]: The file is source via bash and then accessible via a hashmap. This is way faster than using awk/grep for every utilization value. See my comment regarding the performance. https://github.com/green-coding-solutions/eco-ci-energy-estimation/blob/84a01d7b8412a42e2f046f9337aff07a6c4797d3/scripts/make_measurement.sh#L16


That is why the file has this format.

**Pull request merged**